ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_bgp.schema.yml

```diff
@@ -0,0 +1,1500 @@
+                    type: int
+                    min: 1
+                    max: 255
+                    description: Time-to-live in range of hops
+                  next_hop_self:
+                    type: bool
+                  shutdown:
+                    type: bool
+                  bfd:
+                    type: bool
+                  timers:
+                    type: str
+                    description: Time between BGP keepalive messages in seconds "<0-3600> <0-3600>"
+                  rib_in_pre_policy_retain:
+                    type: dict
+                    keys:
+                      enabled:
+                        type: bool
+                      all:
+                        type: bool
+                  send_community:
+                    type: str
+                    description: "'all' or a combination of 'standard', 'extended', 'large' and 'link-bandwidth (w/options)'"
+                    valid_values:
+                      - "all"
+                      - "extended"
+                      - "large"
+                      - "link-bandwidth"
+                      - "standard"
+                  maximum_routes:
+                    type: int
+                    convert_types:
+                      - str
+                  maximum_routes_warning_limit:
+                    type: str
+                    convert_types:
+                      - int
+                    description: |
+                      Maximum number of routes after which a warning is issued (0 means never warn)
+                      Percentage of maximum number of routes at which to warn (1-100)
+                  maximum_routes_warning_only:
+                    type: bool
+                  allowas_in:
+                    type: dict
+                    keys:
+                      enabled:
+                        type: bool
+                      times:
+                        type: int
+                        convert_types:
+                          - str
+                        min: 1
+                        max: 10
+                  default_originate:
+                    type: dict
+                    keys:
+                      enabled:
+                        type: bool
+                      always:
+                        type: bool
+                      route_map:
+                        type: str
+                  update_source:
+                    type: str
+                  route_map_in:
+                    type: str
+                    description: Inbound route-map name
+                  route_map_out:
+                    type: str
+                    description: Outbound route-map name
+                  prefix_list_in:
+                    type: str
+                    description: Inbound prefix-list name
+                  prefix_list_out:
+                    type: str
+                    description: Outbound prefix-list name
+                  address_families:
```

Comment:
Not sure what's going on here... address_families is not a supported data model under vrf.neighbors IMO. It should be under router_bgp.vrfs. Description also doesn't look right, this data model under vrf is not only used for documentation/validation.

Reply:
It is present in structured_config DC1_BL2B.yml not in inventory files, thats why added the key and kept the above description. @carlbuchmann , @ClausHolbechArista please confirm on this.

Reply:
eos_designs group_vars:

https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/DC1_TENANTS_NETWORKS/Tenant_C.yml#L66

structured_config:

https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/structured_configs/DC1-BL2B.yml#L144

Notice the duplicate address_families

```yaml
      neighbors:
        1.1.1.1:
          peer_group: Tenant_C_WAN_Zone_BGP_PEER_GROUP
          description: test_description
          address_families:
            ipv4:
              activate: true
          route_map_out: RM-Tenant_C_WAN_Zone-1.1.1.1-SET-NEXT-HOP-OUT
        BEBA::C0CA:C07A:
          peer_group: Tenant_C_WAN_Zone_BGP_PEER_GROUP
          description: test_ipv6
      address_families:
        ipv4:
          neighbors:
            1.1.1.1:
              activate: true
```

intended config:

https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/DC1-BL2B.cfg#L264

In the current devel branch the issue is that we copy blindly whatever is under `bgp_peers` without checking for keys (which is fine in the code).

https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py#L193

My recommendation here is to remove the address family in the group_vars this should not be supported

Reply:
I believe this is remains from an old version which had more abstraction in eos_cli_config_gen. @Shivani-chourasiya let's remove the key @emilarista pointed to, and remove the invalid key from group_vars.

Reply:
okay, thanks